core/misc/autosubmit.js

```diff
@@ -37,12 +37,39 @@
 Backdrop.behaviors.autosubmit = {
   attach: function(context) {
     // 'this' references the form element
-    function triggerSubmit (e) {
+    function triggerSubmit (element) {
       var $this = $(this);
+      var $triggering_element = $(element);
+
+      // Variable "element" will have a value only when text fields trigger
+      // this. If element is undefined, then remove the data to prevent
+      // potential focus on a previously processed element.
+      if (element === undefined)  {
+        $('body').removeData('needs-focus');
```

Comment:
We should probably namespace this data attribute to more clearly identify that it's coming from `autosubmit.js`. I'd suggest `autosubmit-last-focus-id` to be more descriptive.